File: Core/Ecommerce.Domain/Entities/Order.cs
using Ecommerce.Domain.Common;
using Ecommerce.Shared.Enums;
using System;
using System.Collections.Generic;

namespace Ecommerce.Domain.Entities
{
	public class Order : AuditableEntity
	{
		public int Id { get; set; }
		
		public Guid UserId { get; set; }
		
		public OrderStatus Status { get; set; }
		
		public double Total { get; set; }
		
		public Guid PayPalRequestId { get; set; }
		
		public IEnumerable<OrderItem> OrderItems { get; set; } = null!;
	}
}

File: UI/ecommerce-ui/src/models/Order.ts
import {OrderStatus} from "./OrderStatus.ts";
import {OrderItem} from "./OrderItem.ts";


export class Order {
    id: number;
    userId: string; // Using string to represent Guid
    createdDate: Date;
    status: OrderStatus;
    total: number;
    payPalRequestId: string; // Using string to represent Guid
    orderItems: OrderItem[];

    constructor(
        id: number,
        userId: string,
        createdDate: Date,
        status: OrderStatus,
        total: number,
        payPalRequestId: string,
        orderItems: OrderItem[] = []
    ) {
        this.id = id;
        this.userId = userId;
        this.createdDate = createdDate;
        this.status = status;
        this.total = total;
        this.payPalRequestId = payPalRequestId;
        this.orderItems = orderItems;
    }
}
